### PR TITLE
Add simple live share warning

### PR DESCRIPTION
**Pull request by kerryarchibald**
Includes https://github.com/matrix-org/matrix-react-sdk/pull/8061

https://github.com/matrix-org/matrix-spec-proposals/pull/3489
https://www.figma.com/file/uthEK6xuo9hSQRdWvPAre5/Location-Sharing?node-id=418%3A47576
This is a subset of the complete live location sharing warning banner work.
To be feature + design complete in https://element-io.atlassian.net/browse/PSF-659

Done: 
- [x] Send `m.beacon_info.*` state events.
- [x] track `m.beacon_info.*` events in room state
- [x] expose and maintain '~user~ room has live beacons' property in room state
- [x] add ability to monitor liveness of beacon
- [x] get users own beacons from room state, monitor liveness
- [x] maintain '**user** has live beacons' property)

In this PR:
- [x]  show live state

In next PRs:
- [ ] produce good unique suffixes for beacon info events
- [ ] geolocate and send `m.beacon` events for live beacons
- [ ] maintain liveness state of beacons (checking for expiry, flipping `isLive` on event content)

Screenshot 2022-03-15 at 17 56 43
Screenshot 2022-03-15 at 17 49 57

---
Here's what your changelog entry will look like:

## ✨ Features
 * Add simple live share warning ([\#8066](https://github.com/matrix-org/matrix-react-sdk/pull/8066)). Contributed by @kerryarchibald.

Preview: https://pr8066--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/structures/LoggedInView.tsx [hunk 1]
```diff
@@ -694,8 +695,10 @@ class LoggedInView extends React.Component<IProps, IState> {
                 >
                     <ToastContainer />
                     <div className={bodyClasses}>
-                        <div className='mx_LeftPanel_wrapper'>
-                            { SettingsStore.getValue('TagPanel.enableTagPanel') &&
+                        <div className='mx_LeftPanel_outerWrapper'>
```
Comment (kerryarchibald) [inline src/components/structures/LoggedInView.tsx:698]: Weird diff, change is adding this extra wrapper `mx_LeftPanel_outerWrapper ` and `LeftPanelLiveShareWarning`

Comment (turt2live) [reply src/components/structures/LoggedInView.tsx:698]: looks like the block for `mx_GroupFilterPanelContainer` didn't get indented, which confused the diff a bit. Not that diffs to jsx are ever clear... 😅

Comment (turt2live) [inline src/components/structures/LoggedInView.tsx:698]: in the past we've had performance issues with adding layers to the left panel like this, but it's entirely possible that they have been fixed over the years. Would recommend watching for left panel-related performance complaints post-merge, particularly for large accounts and Chrome users.

**Review by turt2live — APPROVED** [review]

File: src/components/structures/LoggedInView.tsx (hunk 1 above)
Comment (turt2live) [reply src/components/structures/LoggedInView.tsx:698]: surprisingly, it wasn't a performance issue but rather a scrolling issue this time 😭 https://github.com/matrix-org/matrix-react-sdk/pull/8109

**Pull request merged**